maths/prime_number.rb

```diff
@@ -0,0 +1,31 @@
+# A ruby program to check a given number is prime or not
+# Mathematical explanation: A number which has only 2 factors i.e., 1 (one) and itself
+
+# Prime number check function
+def is_prime_number(number)
+  prime_flag = false
+  if number>0
+    if number == 1
+      prime_flag = true
+    else
+      prime_flag = (2..number/2).any? { |i| number % i == 0 }
+    end
+  else
+    prime_flag = true
+  end
+
+  if !prime_flag
+    puts "The given number #{number} is a Prime."
+  else
+    puts "The given number #{number} is not a Prime."
+  end
```

Comment:
**[Optional]**

Generally it is said that methods without side effects (in this case writing a string to standard output) are preferable because they are more testable and composable (ref. [Pure function - Wikipedia](https://en.wikipedia.org/wiki/Pure_function)). It will look like this.

```rb
def prime_number?(number)
  return false if number <= 2

  (3..number.sqrt).step(2).any? { |i| number % i == 0 }
end
```

Note: There are some conventions in naming predicate methods.
- https://rubystyle.guide/#bool-methods-qmark
- https://rubystyle.guide/#bool-methods-prefix